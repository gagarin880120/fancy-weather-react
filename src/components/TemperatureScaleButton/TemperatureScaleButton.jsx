import React from 'react';
import PropTypes from 'prop-types';
import styles from './TemperatureScaleButton.module.css';

export default function TemperatureScaleButton({ scale, onScaleButtonClick, temperatureScale }) {
  return (
    <button
      type="button"
      className={styles.button}
      style={{ border: scale === temperatureScale ? '2px solid #fff' : '2px solid transparent' }}
      onClick={() => onScaleButtonClick(scale)}
    >
      {scale === 'celsius' ? 'C' : 'F'}
    </button>
  );
}

TemperatureScaleButton.propTypes = {
  scale: PropTypes.string,
  onScaleButtonClick: PropTypes.func,
  temperatureScale: PropTypes.string,
};

TemperatureScaleButton.defaultProps = {
  scale: '',
  onScaleButtonClick: null,
  temperatureScale: '',
};
