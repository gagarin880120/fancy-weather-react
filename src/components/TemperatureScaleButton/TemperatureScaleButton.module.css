.button {
  background: transparent;
  color: #fff;
  outline: 0;
  margin-right: 5px;
  width: 35px;
  height: 35px;
  cursor: pointer;
  font-size: 22px;
  font-family: Nunito;
  border-radius: 5px;
  font-weight: bold;
}

.button:active {
  transform: translateY(2px);
}
