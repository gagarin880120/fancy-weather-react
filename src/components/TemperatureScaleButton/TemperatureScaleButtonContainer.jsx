import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import TemperatureScaleButton from './TemperatureScaleButton';
import { setTemperatureScale } from '../../redux/actions';

export function TemperatureScaleButtonContainer({ onScaleButtonClick, scale }) {
  return (
    <TemperatureScaleButton
      onScaleButtonClick={onScaleButtonClick}
      scale={scale}
    />
  );
}

export const mapDispatchToProps = (dispatch) => ({
  onScaleButtonClick(scale) {
    dispatch(setTemperatureScale(scale));
  },
});

TemperatureScaleButtonContainer.propTypes = {
  onScaleButtonClick: PropTypes.func,
};

TemperatureScaleButtonContainer.defaultProps = {
  onScaleButtonClick: null,
};

export default connect(null, mapDispatchToProps)(TemperatureScaleButtonContainer);
