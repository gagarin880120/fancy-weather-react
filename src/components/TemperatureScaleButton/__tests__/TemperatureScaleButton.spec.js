import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TestRenderer, { act } from 'react-test-renderer';
import TemperatureScaleButton from '../TemperatureScaleButton';

describe('TemperatureScaleButton', () => {
  describe('component', () => {
    test('should render without crashing', () => {
      const renderer = new ShallowRenderer();
      renderer.render(
      <TemperatureScaleButton
        scale="fahrenheit"
        temperatureScale="fahrenheit"
      />
      );

      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });

    describe('Events', () => {
      const onScaleButtonClick = jest.fn();

      const testRenderer = TestRenderer.create(
        <TemperatureScaleButton
          scale="celsius"
          onScaleButtonClick={onScaleButtonClick}
        />
      );

      const button = testRenderer.root.findByProps({ type: 'button' });

      describe('onClick', () => {
        test('should call onScaleButtonClick function', () => {

          act(() => {
            button.props.onClick();
          });

          expect(onScaleButtonClick).toHaveBeenCalled();
        });
      });
    });
  });
});
