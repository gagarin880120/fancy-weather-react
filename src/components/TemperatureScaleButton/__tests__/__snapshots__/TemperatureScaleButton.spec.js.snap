// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemperatureScaleButton component should render without crashing 1`] = `
<button
  className="button"
  onClick={[Function]}
  style={
    Object {
      "border": "2px solid #fff",
    }
  }
  type="button"
>
  F
</button>
`;
